src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -2543,8 +2543,14 @@ private void embedGenericHandle(ref CORINFO_RESOLVED_TOKEN pResolvedToken, bool
                     Debug.Assert(pResolvedToken.tokenType == CorInfoTokenKind.CORINFO_TOKENKIND_Method);
                     helperId = ReadyToRunHelperId.MethodDictionary;
                 }
-                
+
+#if READYTORUN
+                MethodDesc exactMethod = (MethodDesc)GetRuntimeDeterminedObjectForToken(ref pResolvedToken);
```

Comment:
Maybe we should just put `GetRuntimeDeterminedObject` into the AOT/R2R specific files and make them return the right thing so that we don't need to ifdef here? (A MethodWithToken for R2R.)

But I admit I don't fully understand why it's okay to forget the token for the type case above and field case below, but not here. It feels like whenever we don't preserve it, it's a bug that we're just not hitting in our "failing test-driven development" model.